pages/index.tsx

```diff
@@ -28,6 +28,27 @@ interface Props {
   variables: MainConfigQueryVariables;
 }
 
+const initialTagsFrequency: Readonly<Record<string, number>> = {};
+
+const calculateSortedTags = (
+  posts: PostDocumentWithoutBody[]
+): readonly TagData[] => {
+  const tagsFrequency = posts
+    .map((post) => post.tags)
+    .flat()
+    .reduce((acc, tag) => {
+      if (acc[tag]) {
+        return { ...acc, [tag]: acc[tag] + 1 };
+      }
+
+      return { ...acc, [tag]: 1 };
+    }, initialTagsFrequency);
+
+  return Object.entries(tagsFrequency)
+    .sort((a, b) => b[1] - a[1])
+    .map(([tagName, postsCount]) => ({ tagName, postsCount }));
+};
+
```

Comment:
This bit of code is a bit hard to read

Try to make it simplier

Reply:
```suggestion
  const calculateSortedTags = (
  posts: PostDocumentWithoutBody[]
): readonly Tag[] => {
  const tags = posts.map((post) => post.tags).flat();

  const tagsFrequency = tags.reduce((acc, tag) => {
    if (acc[tag]) {
      return { ...acc, [tag]: acc[tag] + 1 };
    }

    return { ...acc, [tag]: 1 };
  }, initialTagsFrequency);

  return Object.entries(tagsFrequency)
    .sort((a, b) => b[1] - a[1])
    .map(([name, postsCount]) => ({ name, postsCount }));
};
```